doc/examples/ml/hyperparameter/model.json

```diff
@@ -0,0 +1,48 @@
+{
+  "pipeline": {
+    "name": "model"
+  },
+  "transform": {
+    "image": "pachyderm/iris-train:python-svm-adj-c-gamma",
+    "cmd": [ "/bin/ash" ],
+    "stdin": [
+       "cp /pfs/split/train.csv /tmp/iris.csv",
```

Comment:
I see in a lot of your pipeline specs, that you actually have a decent amount of code in stdin. Is this the correct usage pattern we want to encourage users to do? or should some of this logic (such as the if-statement below) be in the container directly or somewhere else? 

I'm asking more for my own education purposes, but it seems a bit odd to me once you start getting more logic contained in the pipeline spec over the code itself. Overall, it's mostly just setting up directories and naming, so maybe that's fine

Reply:
In my opinion/preference, I like to keep the code as simple as possible `file in -> simple operation -> file out` when possible, then wrap that with some logic about which files and that sets command line arguments. I think this is nice, because it allows you to "version" the different ways you have invoked that simple operation, based on command line args, etc. in previous jobs or versions of the pipeline spec. Also, it prevents hardcoding a bunch of things, and allows you to reuse Docker images more. Again, just my take on things.